framework_tvb/tvb/core/services/import_service.py

```diff
@@ -316,21 +316,29 @@ def import_project_operations(self, project, import_path):
         """
         imported_operations = []
         ordered_operations = self._retrieve_operations_in_order(project, import_path)
+        success_no = 0
 
         for operation_data in ordered_operations:
+
             if operation_data.is_old_form:
                 operation_entity, datatype_group = self.__import_operation(operation_data.operation)
                 new_op_folder = self.files_helper.get_project_folder(project, str(operation_entity.id))
-                operation_datatypes = self._load_datatypes_from_operation_folder(operation_data.operation_folder,
+
+                try:
+                    operation_datatypes = self._load_datatypes_from_operation_folder(operation_data.operation_folder,
                                                                                  operation_entity, datatype_group)
-                # Create and store view_model from operation
-                view_model = self._get_new_form_view_model(operation_entity, operation_data.info_from_xml)
-                h5.store_view_model(view_model, new_op_folder)
-                operation_entity.view_model_gid = view_model.gid.hex
-                dao.store_entity(operation_entity)
+                    # Create and store view_model from operation
+                    view_model = self._get_new_form_view_model(operation_entity, operation_data.info_from_xml)
+                    h5.store_view_model(view_model, new_op_folder)
+                    operation_entity.view_model_gid = view_model.gid.hex
+                    dao.store_entity(operation_entity)
 
-                self._store_imported_datatypes_in_db(project, operation_datatypes)
-                imported_operations.append(operation_entity)
+                    self._store_imported_datatypes_in_db(project, operation_datatypes)
```

Comment:
Only here the dts are to be stored in db, the query to db about possible missing references is much earlier.
Thus, ordering dts by createDate will be necesary But not enough.
Either we check Reference later (in _store_imported_dts_in_db) or we check Reference Both in dao (db) and in the list of dts for the curent operațional (not yet stored in db, only in memory).

Reply:
I think it is okay now.

Reply:
Looks promising. Is the order of dts by create-date happening?
If not, then sometimes it might work, other times not

Reply:
It is happening just before the block of code added by me, at the beginning of the method so I think it should work for all cases.